src/ucp/core/ucp_am.c

```diff
@@ -912,23 +892,25 @@ ucp_am_try_send_short(ucp_ep_h ep, uint16_t id, uint32_t flags,
                       const void *header, size_t header_length,
                       const void *buffer, size_t length)
 {
-    if (ucs_unlikely(((length != 0) && (header_length != 0)) ||
-                     (flags & UCP_AM_SEND_FLAG_RNDV))) {
-        goto out;
-    }
+    size_t total_length;
 
-    if (ucs_unlikely(!ucp_proto_is_inline(ep,
-                                          &ucp_ep_config(ep)->am_u.max_eager_short,
-                                          length + header_length))) {
+    if (ucs_unlikely(flags & UCP_AM_SEND_FLAG_RNDV)) {
         goto out;
     }
 
-    if (!(flags & UCP_AM_SEND_REPLY)) {
-        return ucp_am_send_short(ep, id, flags, header, header_length,
-                                 buffer, length);
-    } else if ((length + header_length) < UCP_AM_SHORT_REPLY_MAX_SIZE) {
-        return ucp_am_send_short_reply(ep, id, flags, header, header_length,
-                                       buffer, length);
+    total_length = header_length + length;
+
+    if (flags & UCP_AM_SEND_REPLY) {
+        if (ucp_proto_is_inline(ep,
```

Comment:
btw,  maybe add `ucs_likely`?

Reply:
here and below

Reply:
it is already present inside `ucp_proto_is_inline`